syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package transport_catalogue;

message Distances {
  string name = 1;
  uint32 distance = 2;
}

message Stop {
  string name = 1;
  double latitude = 2;
  double longitude = 3;
  repeated Distances road_distances = 4;
}

message Bus {
  string name = 1;
  repeated string stop = 2;
  bool is_roundtrip = 3;
}

message TransportCatalogue {
  repeated Bus bus = 1;
  repeated Stop stop = 2;
  RenderSettings render_settings = 3;
  router.Router router = 4;
}
